buildcontext/provider/provider.go

```diff
@@ -67,6 +67,12 @@ func NewBuildContextProvider(console conslogging.ConsoleLogger) *BuildContextPro
 
 // AddDirs adds local directories to the context.
 func (bcp *BuildContextProvider) AddDirs(dirs map[string]string) {
+	for dirName, dir := range dirs {
+		bcp.AddDir(dirName, dir)
+	}
```

Comment:
The real main difference here is that we don't hold the lock the whole time, rather we acquire/release it for each dir. Doesn't appear to be a problem in my cursory testing.